### ISSUE TITLE
Convert BigDecimal to Double by "doubleValue()" method, get a wrong result

### ISSUE THREAD

**Issue by JesseWo** [ionspin/kotlin-multiplatform-bignum#229]
**Describe the bug**

### Unit Test:
```kotlin
  @Test
  fun testToDouble() {
    val result = "29514.9598393574297189".toBigDecimal().doubleValue()
    assertEquals(29514.9598393574297189, result)
  }
```

### Output
```
expected:<29514.95983935743> but was:<0.1693214221471333>
Expected :29514.95983935743
Actual   :0.1693214221471333
```
**To Reproduce**
1. Operation
2. Values to reproduce issue

**Expected behavior**

**Platform**
 - [e.g. JVM, JS, Native, iOS, Linux]

**If JS (please complete the following information):**

**If Smartphone (please complete the following information):**
 - Device: [e.g. Google Pixel 3]
 - OS: [e.g. Android 9.0]
 - Browser [e.g. Chrome]

**Additional context**

**Comment by JesseWo** [comment]
lib version: 0.3.3
1. I have not come across this warning from my Android Studio Chipmunk ｜ 2021.2.1

In addition, the above unit test can be passed if it is reduced by one digit, e.g. `29514.959839357429718`
Thanks for your response!

**Comment by ionspin** [comment]
Hi @JesseWo, I've fixed this issue and the fix will be in build 0.3.6 that is currently rolling out. There was a API change in version 0.3.4 that made `doubleValue()` explicitly require that the exact result is returned, unless you specify that you don't need exact result with `doubleValue(false)`. The reason why we are now requiring is that a lot of people are unaware of the limitations in IEEE 754 double/float (which java and kotlin use)

A simple example would be to run this and observe it finishes successfully even though the numbers are different:
```
assertEquals(29514.9598393574297189, 29514.9598393574297194)
```
